addons/website/controllers/main.py

```diff
@@ -797,6 +799,83 @@ def seo_suggest(self, keywords=None, lang=None):
         xmlroot = ET.fromstring(response)
         return json.dumps([sugg[0].attrib['data'] for sugg in xmlroot if len(sugg) and sugg[0].attrib['data']])
 
+    @http.route(['/website/get_alt_images'], type='jsonrpc', auth="user", website=True)
+    def get_alt_images(self, res_id, res_model):
+        ret = []
+        record = request.env[res_model].browse(res_id)
+        editable_fields = record._editable_fields()
+        if not editable_fields:
+            return ret
+        for editable_field in editable_fields:
+            tree = html.fromstring(str(record[editable_field]))
+            for index, element in enumerate(tree.xpath('//img')):
+                decorative = element.attrib.get('role') == 'presentation'
+                alt = element.attrib.get('alt')
+                if not decorative or not alt:
+                    # If an alt attribute is missing, set decorative as False
+                    # to ensure a correct attribute setting
+                    if not alt:
+                        decorative = False
+                    ret.append({
+                        'id': f"{editable_field}-{index}",
+                        'src': element.attrib.get('src'),
+                        'alt': (element.attrib.get('alt') or '').strip(),
+                        'decorative': decorative,
+                        'updated': False
+                    })
+        return ret
+
+    @http.route(['/website/update_alt_images'], type='jsonrpc', auth="user", website=True)
+    def update_alt_images(self, res_id, res_model, imgs):
+        if not request.env.user.has_group('website.group_website_restricted_editor'):
+            raise werkzeug.exceptions.Forbidden()
+        record = request.env[res_model].browse(res_id)
+        editable_fields = record._editable_fields()
+        if not editable_fields:
+            return False
+        for editable_field in editable_fields:
+            tree = html.fromstring(str(record[editable_field]))
+            modified = False
+            for index, element in enumerate(tree.xpath('//img')):
+                img = next((item for item in imgs if item['id'] == f"{editable_field}-{index}"), None)
+                if not img:
+                    continue
+                if (img['decorative']):
+                    element.set('alt', '')
+                    element.set('role', 'presentation')
+                else:
+                    element.set('alt', markup_escape(img['alt']))
+                    element.attrib.pop('role', None)
+                modified = True
```

Comment:
Is there a case where you don't modify the content ? Normally, the `rpc` is only sent if there is content to modify, right :thinking: ?

Reply:
Yes, that's true in most cases. However, the model can have multiple fields (as indicated by 'editable_fields'), so it's possible for a broken link to exist in only one specific field, even if others remain unchanged.